lib/elastomer/client/scroller.rb

```diff
@@ -120,7 +117,25 @@ def clear_scroll( scroll_ids )
       response.body
     end
 
-    DEFAULT_OPTS = {
+    # Internal: Add sort by doc to query.
+    #
+    # Raises an exception if the query contains a sort already.
+    # Returns the query as a hash
+    def add_sort_by_doc(query)
+      if query.nil?
+        query = {}
+      elsif query.is_a? String
+        query = MultiJson.load(query)
+      end
+
+      if query.has_key? :sort
+         raise ArgumentError, "#{query} cannot contain a sort #{query[:sort]}"
```

Comment:
Since query could be quite large I think this exception is going to be hard to read. I'd suggest something like:

`Query cannot contain a sort (found sort '#{query[:sort]}' in query: #{query})`

With the query on the end.

Reply:
Great suggestion, thanks.